tests/d3d12_command.c

```diff
@@ -490,6 +490,197 @@ void test_fractional_viewports(void)
+            {{ 0.0f, 0.0f, -0.5f, 1.0f}, {0.2f, 0.2f}},
+        },
+        {
+            {{0.0f,  0.0f, 0.5f, 1.0f}, {0.3f, 0.3f}},
+            {{0.0f, -1.0f, 0.5f, 1.0f}, {0.3f, 0.3f}},
+            {{ 1.0f,  0.0f, 0.5f, 1.0f}, {0.3f, 0.3f}},
+            {{ 1.0f, -1.0f, 0.5f, 1.0f}, {0.3f, 0.3f}},
+        },
+        {
+            {{-1.0f,  0.0f, -1.0f, 1.0f}, {0.4f, 0.4f}},
+            {{-1.0f, -1.0f, -1.0f, 1.0f}, {0.4f, 0.4f}},
+            {{ 0.0f,  0.0f, -1.0f, 1.0f}, {0.4f, 0.4f}},
+            {{ 0.0f, -1.0f, -1.0f, 1.0f}, {0.4f, 0.4f}},
+        },
+    };
+
+    memset(&desc, 0, sizeof(desc));
+    desc.rt_format = DXGI_FORMAT_R32G32B32A32_FLOAT;
+    desc.no_root_signature = true;
+    if (!init_test_context(&context, &desc))
+        return;
+    command_list = context.list;
+    queue = context.queue;
+    device = context.device;
+
+    memset(&options13, 0, sizeof(options13));
+    hr = ID3D12Device_CheckFeatureSupport(device, D3D12_FEATURE_D3D12_OPTIONS13, &options13, sizeof(options13));
+    ok(SUCCEEDED(hr), "OPTIONS13 is not supported by runtime.\n");
+
+    if (!options13.InvertedViewportHeightFlipsYSupported)
+    {
+        skip("Device does not support negative viewport height.\n");
+        destroy_test_context(&context);
+        return;
+    }
+
+    context.root_signature = create_empty_root_signature(context.device,
+            D3D12_ROOT_SIGNATURE_FLAG_ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT);
+
+    input_layout.pInputElementDescs = layout_desc;
+    input_layout.NumElements = ARRAY_SIZE(layout_desc);
+
+    context.pipeline_state = create_pipeline_state(context.device,
+            context.root_signature, desc.rt_format, &vs, &ps, &input_layout);
+
+    for (i = 0; i < 4; i++)
+    {
+        vbs[i] = create_upload_buffer(context.device, sizeof(quads[i]), quads[i]);
+
+        vbvs[i].BufferLocation = ID3D12Resource_GetGPUVirtualAddress(vbs[i]);
+        vbvs[i].StrideInBytes = sizeof(quads[i][0]);
+        vbvs[i].SizeInBytes = sizeof(quads[i]);
+    }
+
+    set_viewport(&viewport, 0, context.render_target_desc.Height,
+            context.render_target_desc.Width, -(float)context.render_target_desc.Height, 1.0f, 0.0f);
+
+    ID3D12GraphicsCommandList_ClearRenderTargetView(command_list, context.rtv, white, 0, NULL);
+
+    ID3D12GraphicsCommandList_OMSetRenderTargets(command_list, 1, &context.rtv, false, NULL);
+    ID3D12GraphicsCommandList_SetGraphicsRootSignature(command_list, context.root_signature);
+    ID3D12GraphicsCommandList_SetPipelineState(command_list, context.pipeline_state);
+    ID3D12GraphicsCommandList_IASetPrimitiveTopology(command_list, D3D_PRIMITIVE_TOPOLOGY_TRIANGLESTRIP);
+    ID3D12GraphicsCommandList_RSSetViewports(command_list, 1, &viewport);
+    ID3D12GraphicsCommandList_RSSetScissorRects(command_list, 1, &context.scissor_rect);
+
+    for (i = 0; i < 4; i++)
+    {
+        ID3D12GraphicsCommandList_IASetVertexBuffers(command_list, 0, 1, &vbvs[i]);
```

Comment:
I meant that we need 4 different RSSetViewports here rather than using the VBO to do the same thing.

Reply:
Done, PTAL.